imap_processing/ialirt/l0/process_hit.py

```diff
@@ -0,0 +1,210 @@
+"""Functions to support HIT processing."""
+
+import logging
+from enum import Enum
+from typing import ClassVar
+
+import numpy as np
+import xarray as xr
+
+logger = logging.getLogger(__name__)
+
+
+class HITPrefixes(Enum):
+    """Create ENUM for each rate type."""
+
+    FAST_RATE_1: ClassVar[list[str]] = [
+        f"{prefix}_{i:02d}"
+        for i in range(15)
+        for prefix in ["L1A_TRIG", "IA_EVNT_TRIG", "A_EVNT_TRIG", "L3A_TRIG"]
+    ]
+    FAST_RATE_2: ClassVar[list[str]] = [
+        f"{prefix}_{i:02d}"
+        for i in range(15)
+        for prefix in ["L1B_TRIG", "IB_EVNT_TRIG", "B_EVNT_TRIG", "L3B_TRIG"]
+    ]
+    SLOW_RATE: ClassVar[list[str]] = [
+        "L1A",
+        "L2A",
+        "L3A",
+        "L1A0AHG",
+        "L1B0AHG",
+        "L1C0AHG",
+        "L4IAHG",
+        "L4OAHG",
+        "SLOW_RATE_08",
+        "SLOW_RATE_09",
+        "SLOW_RATE_10",
+        "L1A0BHG",
+        "L1B0BHG",
+        "L1C0BHG",
+        "L4IBHG",
+        "L4OBHG",
+        *[f"IALRT_RATE_{i}" for i in range(1, 21)],
+        "TRIG_IA_EVNT",
+        "TRIG_IB_EVNT",
+        "NASIDE_IALRT",
+        "NBSIDE_IALRT",
+        *[f"ERATE_{i}" for i in range(1, 6)],
+        "L12A",
+        "L123A",
+        "PENA",
+        "L12B",
+        "L123B",
+        "PENB",
+        "SLOW_RATE_51",
+        "SLOW_RATE_52",
+        "SLOW_RATE_53",
+        "SLOW_RATE_54",
+        "H_06_08",
+        "H_12_15",
+        "H_15_70",
+        "HE4_06_08",
+        "HE4_15_70",
+    ]
+
+
+def find_groups(data: xr.Dataset) -> xr.Dataset:
+    """
+    Find groups in the data.
+
+    Parameters
+    ----------
+    data : xr.Dataset
+        HIT Dataset.
+
+    Returns
+    -------
+    grouped_data : xr.Dataset
+        Grouped data.
+    """
+    subcom_range = (0, 59)
+
+    data = data.sortby("hit_sc_tick", ascending=True)
+
+    # Filter out data before the first subcom=0 and after the last subcom=59
+    start_sc_ticks = data["hit_sc_tick"][(data["hit_subcom"] == subcom_range[0]).values]
+    start_sc_tick = start_sc_ticks.data.min()
+    last_sc_ticks = data["hit_sc_tick"][([data["hit_subcom"] == subcom_range[-1]][-1])]
+    last_sc_tick = last_sc_ticks.data.max()
+
+    grouped_data = data.where(
+        (data["hit_sc_tick"] >= start_sc_tick) & (data["hit_sc_tick"] <= last_sc_tick),
+        drop=True,
+    )
+
+    # Create group labels
+    group_labels = np.searchsorted(
+        start_sc_ticks, grouped_data["hit_sc_tick"], side="right"
+    )
+    grouped_data["group"] = ("group", group_labels)
```

Comment:
Could you add an example of what `grouped_data` looks like? I'm not following all of the logic here and think you might just be getting one "group"? Or are you labeling every item in the array with a "group value", like this set of `[0-59]` all has an attached "1" label, the next set of `[0-59]` has a "2" label, etc.